Manak_Leah_ndvi_2022.py

```diff
@@ -0,0 +1,670 @@
+
+    Parameters
+    -----------
+    directory_path : string
+        A path to the directory name
+    sitename_location : index list
+        Index of sitename location in directory path name
+    datetime_location : index list
+        Index of datetime location in directory path name
+
+    Returns
+    -----------
+    list : list of the datetime location and sitename information
+    """
+    # Create an empty list to append both sitename and date information
+    sitename_date = []
+
+    # Assign datetime location to an object and specify datetime format
+    date_location = directory_path[datetime_location[0]: datetime_location[1]]
+    format = "%Y%m%d"
+
+    # Create a date varaiable using new object and the datetime format
+    date = datetime.strptime(date_location, format)
+
+    # Create a location variable called "site"
+    site = directory_path[sitename_location[0]: sitename_location[1]]
+
+    # Append site and date variables to list
+    sitename_date.append(site)
+    sitename_date.append(date)
+    
+    # Return the populated sitename_date list
+    return sitename_date
+
+# A function to open clean landsat bands
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None):
+    """Open, crop, and identify the range of the bands.
+
+    Parameters
+    -----------
+    band_path : string
+        A path to the array that we will open
+    valid_range : tuple (optional)
+        A tuple of min and max range of values for the data. Default = None
+
+    Returns
+    -----------
+    arr : xarray DataArray
+        An xarray DataArray with values that should be masked
+        set to 1 for True (Boolean)
+    """
+    # tests to ensure the arrays are clipped to the same .shape
+    band = rxr.open_rasterio(
+        band_path, masked=True).rio.clip(
+        crop_extent.geometry,from_disk=True).squeeze()
+
+    # This last step is only for a valid tuple
+    if valid_range:
+        mask = ((band < valid_range[0]) | (band > valid_range[1]))
+        band = band.where(~xr.where(mask, True, False))
+
+    return band
+
+
+# In[6]:
+
+
+# Create dataframe of mean NDVI in this cell using the functions created above
+
+# Create path to the two sites "SJER" and "HARV"
+site_paths = glob(os.path.join("ndvi-automation",
+                               "sites",
+                               "*"))
+site_paths
+
+# Create cloud mask and get the cloud pixel values from earthpy
+high_cloud_confidence = (
+    em.pixel_flags["pixel_qa"]["L8"]["High Cloud Confidence"])
+cloud = em.pixel_flags["pixel_qa"]["L8"]["Cloud"]
+cloud_shadow = em.pixel_flags["pixel_qa"]["L8"]["Cloud Shadow"]
+
+all_masked_values = cloud_shadow + cloud + high_cloud_confidence
+harv_path = glob(os.path.join("ndvi-automation", "sites", "HARV"))
+
+# Open and clean all HARV: first create empty HARV list
+harv_info = []
+
+# Create a for loop for the HARV path
+for path in harv_path:
```

Comment:
@LManak since we're just accessing one HARV site scene, I wonder if you could do the bulk of this processing without any, or less, for loops. Could you extract just the specific files you need to process, and then run your functions on those? 

Multiple nested for loops tend to have really [bad time complexities](https://towardsdatascience.com/understanding-time-complexity-with-python-examples-2bda6e8158a7), in this case since there are three it's O(n^3). So it's always best to minimize these as much as you can!